shopfloor/models/stock_move.py

```diff
@@ -36,3 +36,42 @@ def _action_done(self, cancel_backorder=False):
                 if picking.state == "done":
                     picking.action_done()
         return moves
+
+    def extract_and_action_done(self):
+        """Extract the moves in a separate transfer and validate them.
+
+        You can combine this method with `split_other_move_lines` method
+        to first extract some move lines in a separate move, then validate it
+        with this method.
+        """
+        moves = self.filtered(lambda m: m.state == "assigned")
+        if not moves:
+            return False
+        for picking in moves.picking_id:
+            moves_todo = picking.move_lines & moves
+            if moves_todo == picking.move_lines:
+                # No need to create a new transfer if we are processing all moves
+                new_picking = picking
+            else:
+                new_picking = picking.copy(
+                    {
+                        "name": "/",
+                        "move_lines": [],
+                        "move_line_ids": [],
+                        "backorder_id": picking.id,
+                    }
+                )
+                new_picking.message_post(
+                    body=_(
+                        "Created from backorder "
+                        "<a href=# data-oe-model=stock.picking data-oe-id=%d>%s</a>."
+                    )
+                    % (picking.id, picking.name)
+                )
+            moves_todo.write({"picking_id": new_picking.id})
+            moves_todo.mapped("package_level_id").write({"picking_id": new_picking.id})
+            moves_todo.mapped("move_line_ids").write({"picking_id": new_picking.id})
```

Comment:
Worth to add
```python
moves_todo.move_line_ids.package_level_id.write(
    {"picking_id": new_picking.id}
)
```
As done here: https://github.com/camptocamp/wms/blob/13.0-shopfloor/shopfloor/models/stock_move_line.py#L103-L107

Reply:
added